hazelcast/src/main/java/com/hazelcast/map/impl/AbstractEvictableRecordStore.java

```diff
@@ -382,6 +383,16 @@ protected void accessRecord(Record record, long now) {
         setExpirationTime(record, maxIdleMillis);
     }
 
+    protected void mergeRecordExpiration(Record record, EntryView mergingEntry) {
+        if (mergingEntry.getTtl() > 0 && mergingEntry.getTtl() != Long.MAX_VALUE) {
```

Comment:
If `TTL = 0`, entry should live eternally in `IMap`. Can you please state your intention by not setting zero TTL from merging entry?